test/routes/assignment_cycle_process.js

```diff
@@ -0,0 +1,195 @@
+    1
+  );
+
+  const preference21 = await create_models.create_preference(
+    chore2.id,
+    user1.id,
+    1
+  );
+  const preference22 = await create_models.create_preference(
+    chore2.id,
+    user2.id,
+    2
+  );
+  const preference23 = await create_models.create_preference(
+    chore2.id,
+    user3.id,
+    3
+  );
+
+  const preference31 = await create_models.create_preference(
+    chore3.id,
+    user1.id,
+    3
+  );
+  const preference32 = await create_models.create_preference(
+    chore3.id,
+    user2.id,
+    1
+  );
+  const preference33 = await create_models.create_preference(
+    chore3.id,
+    user3.id,
+    2
+  );
+
+  const preference41 = await create_models.create_preference(
+    chore4.id,
+    user1.id,
+    2
+  );
+  const preference42 = await create_models.create_preference(
+    chore4.id,
+    user2.id,
+    0
+  );
+  const preference43 = await create_models.create_preference(
+    chore4.id,
+    user3.id,
+    0
+  );
+
+  t.truthy(await PreferenceChecker.validateUserPreferencesByRoomId(room1.id));
+
+  const res1 = await request(app)
+    .post("/assignment/assignmentCycle")
+    .expect(200);
+
+  //Should have 4 assignments, 5 chores, 4 active, 2 upcoming,
+  const assignments1 = await Assignment.find({});
+  const activeAssignments1 = await Assignment.find({ active: true });
+  const chores1 = await Chore.find({});
+  const upcomingChores1 = await Chore.find({ upcoming: true });
+  const activeChores1 = await Chore.find({ active: true });
+
+  t.truthy(await PreferenceChecker.validateUserPreferencesByRoomId(room1.id));
+  t.truthy(assignments1.length == 4);
+  t.truthy(activeAssignments1.length == 4);
+  t.truthy(chores1.length == 5);
+  t.truthy(upcomingChores1.length == 2);
+  t.truthy(activeChores1.length == 4);
+  t.truthy(await PreferenceChecker.validateUserPreferencesByRoomId(room1.id));
+
+  //PART 2
+
+  const chore6 = await create_models.create_chore(room1.id, "CUPS"); //Upcoming, non recurring
+  const preference61 = await create_models.create_preference(
+    chore6.id,
+    user1.id,
+    53
+  ); //Test preference checking/fixing
+  const preference62 = await create_models.create_preference(
+    chore6.id,
+    user2.id,
+    32
+  );
+  const preference63 = await create_models.create_preference(
+    chore6.id,
+    user3.id,
+    23
+  );
+
+  const res2 = await request(app)
+    .post("/assignment/assignmentCycle")
+    .expect(200);
+
+  //should have 7 assignments in total, 3 active, 7 nonsuccess, 6 chores, 2 upcoming, 3 inactive(2 upcoming are recurring)
+  const assignments2 = await Assignment.find({});
+  const inactiveAssignments2 = await Assignment.find({ active: false });
+  const nonsuccessAssignments2 = await Assignment.find({ successful: false });
+  const chores2 = await Chore.find({});
+  const inactiveChores2 = await Chore.find({ active: false });
+  const upcomingChores2 = await Chore.find({ upcoming: true });
+
+  t.truthy(assignments2.length == 7);
```

Comment:
`t.is(assignments2.length, 7)` is a stronger assertions than truthy. You should probably change the rest of assertions to make your tests stronger.